src/main/java/seedu/address/logic/parser/FindBillCommandParser.java

```diff
@@ -61,10 +72,39 @@ public FindBillCommand parse(String args) throws ParseException {
                         String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindBillCommand.MESSAGE_USAGE));
             }
 
-            this.predicate = new PaymentStatusPredicate(trimmedArgs);
+            this.paymentStatusPredicate = (paymentStatus -> paymentStatus.toString().equals(trimmedArgs));
         }
 
-        return new FindBillCommand(this.predicate);
+        Optional<Predicate<PaymentStatus>> finalPaymentStatusPredicate =
+                Optional.ofNullable(this.paymentStatusPredicate);
+
+        if (argMultimap.getValue(PREFIX_BILL_DATE).isPresent()) {
+            String trimmedArgs = argMultimap.getValue(PREFIX_BILL_DATE).get().trim();
+
+            if (!trimmedArgs.matches("^[0-9:-]+$")) {
+                throw new ParseException("Only numbers, - and : are allowed as input for finding by slot");
+            }
+
+            this.billDatePredicate = (billDate -> billDate.toString().contains(trimmedArgs.toLowerCase()));
+        }
+
+        Optional<Predicate<BillDate>> finalBillDatePredicate = Optional.ofNullable(this.billDatePredicate);
+
+        if (argMultimap.getValue(PREFIX_AMOUNT).isPresent()) {
+            String trimmedArgs = ParserUtil.parseAmount(argMultimap
+                    .getValue(PREFIX_AMOUNT).get()).toString().trim();
+            if (trimmedArgs.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindBillCommand.MESSAGE_USAGE));
+            }
+
+            this.amountPredicate = (amount -> amount.toString().equals(trimmedArgs));
```

Comment:
use contains instead of equals, can check regex using ^[0-9.]+$ which only allows for numbers and decimal point .